source/extensions/common/wasm/wavm/wavm.cc

```diff
@@ -281,10 +286,9 @@ bool Wavm::initialize(const std::string& code, absl::string_view name, bool allo
     rootResolver.moduleNameToInstanceMap().set("global", emscriptenInstance_->global);
   }
```

Comment:
The whole `needs_emscripten` logic can be now replaced with `is_emscripten_`.

Reply:
So I can do that, but the user-defined sections are not supported in WAT, at least not by wavm-dis so it might be best to drop WAT support, particularly since v8 doesn't have it.

Reply:
Worked around it.  We can remove support in another PR.